versioned/persist/tx/src/test/java/org/projectnessie/versioned/persist/tx/AbstractTxDatabaseAdapterTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.tx;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.tuple;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+import org.junit.jupiter.api.Test;
+import org.projectnessie.versioned.persist.tests.AbstractDatabaseAdapterTest;
+import org.projectnessie.versioned.persist.tx.TxDatabaseAdapter.NessieSqlDataType;
+
+public abstract class AbstractTxDatabaseAdapterTest extends AbstractDatabaseAdapterTest {
+  @Test
+  void isIntegrityConstraintViolation() throws Exception {
```

Comment:
nit: rename to `insertOnConflict` perhaps (for clarity)?

Reply:
:+1: